spec/controllers/api/v1/accounts/articles_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+RSpec.describe 'Api::V1::Accounts::Articles', type: :request do
+  let(:account) { create(:account) }
+  let(:agent) { create(:user, account: account, role: :agent) }
+  let!(:portal) { create(:portal, name: 'test_portal', account_id: account.id) }
+  let!(:category) { create(:category, name: 'category', portal: portal, account_id: account.id, locale: 'en') }
+  let!(:article) { create(:article, category: category, portal: portal, account_id: account.id, author_id: agent.id) }
+
+  describe 'POST /api/v1/accounts/{account.id}/portals/{portal.slug}/articles' do
+    context 'when it is an unauthenticated user' do
+      it 'returns unauthorized' do
+        post "/api/v1/accounts/#{account.id}/portals/#{portal.slug}/articles", params: {}
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when it is an authenticated user' do
+      it 'creates article' do
```

Comment:
If the user is of agent role, then they should be a portal member to access it. 

If this case is not handled here, let us pick it as a separate ticket.